From 2870e908ce733e6f249aa6dd388a365c7a2a4425 Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Thu, 4 Jun 2015 20:36:40 +0000
Subject: [PATCH] readlink: simplify -f

Use output from realpath(3) directly. This makes usage
with . more consistent with other implementations.
---
 readlink.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/readlink.c b/readlink.c
index 3fa1f2e34d8a..a72390ffc737 100644
--- a/readlink.c
+++ b/readlink.c
@@ -76,11 +76,8 @@ mdone:
 			eprintf("realpath %s:", argv[0]);
 		break;
 	case 'f':
-		p = dirname(estrdup(argv[0]));
-		if (!realpath(p, b))
-			eprintf("realpath %s:", p);
-		estrlcat(b, "/", sizeof(arg));
-		estrlcat(b, basename(estrdup(argv[0])), sizeof(arg));
+		if (!realpath(argv[0], b))
+			eprintf("realpath %s:", argv[0]);
 		break;
 	default:
 		if ((n = readlink(argv[0], b, PATH_MAX - 1)) < 0)
-- 
2.4.2

